### DISCUSSION TITLE
vpype read attribute color

### DISCUSSION THREAD

**Discussion by willyk26** [abey79/vpype#574] [Ideas]
the read attribute stroke parameter is able to parse color information and accurately assign a color to an individual layer, would it be a challenge to implement a feature that also names that layer to the corresponding color so that it could then be managed by a unique pen configuration?

**Comment by abey79** [comment]
The question is about naming a layer based on its color? You can do that with expressions, e.g.:

```bash
 vpype read -a stroke test.svg forlayer name -l %_lid% %_prop.svg_stroke% end show
```

With the SVG below, I get the following result:
<img width="772" alt="image" src="https://user-images.githubusercontent.com/49431240/199211969-6d25aa57-506f-4ec9-b25b-b163d4a7c767.png">

The SVG:

![test](https://user-images.githubusercontent.com/49431240/199212081-dc7dda31-506f-43a6-9f74-aeaf7dea68cb.svg)
